### PR TITLE
ROX-14267: Allow central to `get pods/log`

### PR DESCRIPTION
**Pull request by 0x656b694d**
## Description

* Added error debug message for the case when logs cannot be retrieved;
* Added "get pods/log" permission to the `stackrox-central-diagnostic` role.

## Checklist
- [x] Investigated and inspected CI test results
- [ ] ~Unit test and regression tests added~
- [ ] ~Evaluated and added CHANGELOG entry if required~
- [ ] ~Determined and documented upgrade steps~
- [ ] ~Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

Tested with local setup.

Before giving permissions:
```
Error retrieving container logs: unknown (get pods central-756858c7cf-6wgts)
server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "unknown (get pods central-756858c7cf-6wgts)",
  "reason": "Forbidden",
  "details": {
    "name": "central-756858c7cf-6wgts",
    "kind": "pods",
    "causes": [
      {
        "reason": "UnexpectedServerResponse",
        "message": "unknown"
      }
    ]
  },
  "code": 403
}]
```

### CODE REVIEW

**Comment by 0x656b694d** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6373** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6373" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6373?utm_source=stack-comment).

File: pkg/k8sintrospect/collector.go [hunk 1]
```diff
@@ -183,7 +183,8 @@ func (c *collector) collectPodData(pod *v1.Pod) error {
 			}
 			logsData, err := c.client.CoreV1().Pods(pod.GetNamespace()).GetLogs(pod.GetName(), podLogOpts).DoRaw(c.ctx)
 			if err != nil {
-				logsData = []byte(fmt.Sprintf("Error retrieving container logs: %v", err))
+				logsData = []byte(fmt.Sprintf("Error retrieving container logs: %v\n", err))
```
Comment (ivan-degtiarenko) [inline pkg/k8sintrospect/collector.go:186]: Perhaps we can also specify `container.Name` in the error message

File: pkg/k8sintrospect/collector.go [hunk 2]
```diff
@@ -208,7 +209,8 @@ func (c *collector) collectPodData(pod *v1.Pod) error {
 			}
 			logsData, err := c.client.CoreV1().Pods(pod.GetNamespace()).GetLogs(pod.GetName(), podLogOpts).DoRaw(c.ctx)
 			if err != nil {
-				logsData = []byte(fmt.Sprintf("Error retrieving previous container logs: %v", err))
+				logsData = []byte(fmt.Sprintf("Error retrieving previous container logs: %v\n", err))
```
Comment (ivan-degtiarenko) [inline pkg/k8sintrospect/collector.go:212]: Same here

**Review by ivan-degtiarenko — APPROVED** [review]
LGTM modulo 2 suggestions

**Pull request merged**